bindings/guile/gnc-kvp-guile.cpp

```diff
@@ -21,6 +22,17 @@ extern "C"
  *       types based only on the scheme type.
  */
 
+static bool scm_is_list_of_string_pairs (SCM val)
+{
+ restart:
+    if (scm_is_null (val)) return true;
```

Comment:
Not the first time through!

Reply:
This comes from scheme... An empty list is still a list.

Reply:
It might be, but it's not a list of string pairs so you can't deduce that the KvpValue type should be a KvpFrame; it might as easily be a GList.